src/extensions/checkProjectBilling.ts

```diff
@@ -0,0 +1,130 @@
+import * as clc from "cli-color";
+import * as opn from "open";
+
+import * as cloudbilling from "../gcp/cloudbilling";
+import { FirebaseError } from "../error";
+import { logger } from "../logger";
+import { logPrefix } from "./extensionsHelper";
+import * as prompt from "../prompt";
+import * as utils from "../utils";
+
+/**
+ * Logs to console if setting up billing was successful.
+ */
+function logBillingStatus(enabled: boolean, projectId: string): void {
+  if (!enabled) {
+    throw new FirebaseError(
+      `${logPrefix}: ${clc.bold(
+        projectId
+      )} could not be upgraded. Please add a billing account via the Firebase console before proceeding.`
+    );
+  }
+  utils.logLabeledSuccess(logPrefix, `${clc.bold(projectId)} has successfully been upgraded.`);
+}
+
+/**
+ * Opens URL if applicable and stalls until user responds.
+ */
+async function openBillingAccount(projectId: string, url: string, open: boolean): Promise<boolean> {
+  if (open) {
+    try {
+      opn(url);
+    } catch (err) {
+      logger.debug("Unable to open billing URL: " + err.stack);
+    }
+  }
+
+  await prompt.promptOnce({
+    name: "continue",
+    type: "confirm",
+    message:
+      "Press enter when finished upgrading your project to continue setting up your extension.",
+    default: true,
+  });
+  return cloudbilling.checkBillingEnabled(projectId);
+}
+
+/**
+ * Question prompts user to select billing account for project.
+ */
+async function chooseBillingAccount(
+  projectId: string,
+  extensionName: string,
+  accounts: cloudbilling.BillingAccount[]
+): Promise<void> {
+  const choices = accounts.map((m) => m.displayName);
+  choices.push("Add new billing account");
+
+  const answer = await prompt.promptOnce({
+    name: "billing",
+    type: "list",
+    message: `The extension ${clc.underline(
+      extensionName
+    )} requires your project to be upgraded to the Blaze plan. You have access to the following billing accounts.
+Please select the one that you would like to associate with this project:`,
+    choices: choices,
+  });
+
+  let billingEnabled: boolean;
+  if (answer === "Add new billing account") {
```

Comment:
Should this be a constant?

Reply:
I don't think so? We setto something different in both the if and else block, and then use it after. Do you see a cleaner way to write this chunk that keeps it as a const?

Reply:
Oh, I meant the string "Add new billing account" could be a constant defined at the top of the file. Since this is an option that the user selects, not a string they enter. Also if the string option changes, we have to remember to update it in two places.

Reply:
Oh, i totally misread which line this comment was referring to 🤦 , good point.